### ISSUE TITLE
No FD found to solve Objective o_navigateA

### ISSUE THREAD

**Issue by chcorbato** [rosin-project/metacontrol_sim#50]
This happens in 35cd0dc when using the updated launchfile:
```console
$ roslaunch metacontrol_sim metacontrol_sim_reasoner.launch
[...]
[INFO] [1600862917.455973, 7.662000]: QA value received!        TYPE: safety    VALUE: 0.490094699729

FGs:
fg_f3_v3_r1     objective:  kb.o_navigateA      status:  None   FD:  kb.f3_v3_r1        QAvalues:  [('safety', 0.490094699729), ('energy', 0.415905276934)]

OBJECTIVE       |  STATUS       |  NFRs
[WARN] [1600862917.457630, 7.663000]: QA value received for             TYPE: safety    VALUE: 0.702318014465
o_navigateA     |   INTERNAL_ERROR      |   [('energy', 0.5), ('safety', 0.8)]
[WARN] [1600862917.458305, 7.663000]: Objectives in status ERROR: ['o_navigateA']
[INFO] [1600862917.460224, 7.663000]: QA value received!        TYPE: safety    VALUE: 0.702318014465
[WARN] [1600862917.461828, 7.663000]: QA value received for             TYPE: safety    VALUE: 0.79356236787
[INFO] [1600862917.466717, 7.664000]:   >> Finished MAPE-K ** ANALYSIS **
[INFO] [1600862917.468758, 7.666000]:   >> Started MAPE-K ** PLAN adaptation **
[INFO] [1600862917.470665, 7.666000]: => Reasoner searches FD for objective: o_navigateA
[INFO] [1600862917.471677, 7.666000]: QA value received!        TYPE: safety    VALUE: 0.79356236787
== FunctionDesigns available for obj: %s ['f1_v1_r1', 'f1_v1_r2', 'f1_v1_r3', 'f1_v2_r1', 'f1_v2_r2', 'f1_v2_r3', 'f1_v3_r1', 'f1_v3_r2', 'f1_v3_r3', 'f2_v1_r1', 'f2_v1_r2', 'f2_v1_r3', 'f2_v2_r1', 'f2_v2_r2', 'f2_v2_r3', 'f2_v3_r1', 'f2_v3_r2', 'f2_v3_r3', 'f3_v1_r1', 'f3_v1_r2', 'f3_v1_r3', 'f3_v2_r1', 'f3_v2_r2', 'f3_v2_r3', 'f3_v3_r1', 'f3_v3_r2', 'f3_v3_r3']
Objective NFR ENERGY: %s [kb.nfr_energy, kb.nfr_safety]
== Checking FDs for Objective with NFRs type: %s and value %s  energy 0.5
No FDs meetf NFRs
*** OPERATOR NEEDED, NO SOLUTION FOUND ***
[ERROR] [1600862917.475400, 7.668000]: No FD found to solve Objective o_navigateA
[...]
```

Maybe it is due to the particular scenario used in this launchfile, can you please check @marioney

**Comment by marioney** [comment]
> Maybe it is due to the particular scenario used in this launchfile, can you please check @marioney

This should be solved with https://github.com/rosin-project/metacontrol_sim/pull/51 and https://github.com/tud-cor/mc_mros_reasoner/commit/015428688ced89b94eac59fb4841aaabf9a118ff

Can you check if this works for you?

**Comment by chcorbato** [comment]
Confirmed, it is solved with #51 and tud-cor/mc_mros_reasoner@0154286
